lua/pac3/editor/server/wear.lua

```diff
@@ -298,33 +294,58 @@ function pace.SubmitPart(data, filter)
 	return true
 end
 
+-- Inserts the given part into the StreamQueue
+function pace.SubmitPart(data, filter, callback)
+	if istable(data.part) then
+		pac.dprint("queuing part %q from %s", data.part.self.Name, tostring(data.owner))
+		table.insert(pace.StreamQueue, {
+			data = data,
+			filter = filter,
+			callback = callback
+		})
+
+		return "queue"
+	end
+
+	return pace.SubmitPartNow(data, filter)
+end
```

Comment:
The new `SubmitPart` function with two key differences:
1. It takes a third `callback` param
2. It will always return `"queue"` (to maintain compatibility with the old function)

So this function will never actually call the real worker function immediately.